flink-core/src/main/java/org/apache/flink/metrics/reporter/AbstractReporter.java

```diff
@@ -84,4 +84,24 @@ public void notifyOfRemovedMetric(Metric metric, String metricName, AbstractMetr
 	protected String replaceInvalidChars(String metricName) {
 		return metricName;
 	}
+
+	/**
+	 * Method which constructs the fully qualified metric name from the metric group and the metric
+	 * name.
+	 *
+	 * @param metricName Name of the metric
+	 * @param group Associated metric group
+	 * @return Fully qualified metric name
+	 */
+	private String constructMetricName(String metricName, AbstractMetricGroup group) {
+		StringBuilder builder = new StringBuilder();
+
+		for (String componentName : group.getScopeComponents()) {
+			builder.append(replaceInvalidChars(componentName)).append(".");
+		}
+
+		builder.append(replaceInvalidChars(metricName));
```

Comment:
the call to replaceInvalidChars is not required; metric names can only contain alphanumeric characters. I don't think any reporter will not support these.